### PR TITLE
Fix chat_message..js bracketing

### PR DESCRIPTION
**Pull request by Nurutomo**

### CODE REVIEW

File: lib/chat_message.js [hunk 1]
```diff
@@ -260,8 +260,9 @@ class ChatMessage {
     }
 
     let message = Object.keys(codes).map((code) => {
-      if (!this[code] || this[code] === 'false') return null
-      if (code === 'color') return codes.color[this.color]
+      self[code] = parent[code] || this[code]
```
Comment (wvffle) [inline lib/chat_message.js:263]: Self is not defined.

File: lib/chat_message.js [hunk 2]
```diff
@@ -260,8 +260,9 @@ class ChatMessage {
...
+      if (!self[code] || self[code] === 'false') return null
+      if (code === 'color') return codes.color[self.color]
```
Comment (wvffle) [inline lib/chat_message.js:265]: if you could, please add brackets to improve readability
```js
if (...) { 
  ...
}
```

File: lib/chat_message.js [hunk 3]
```diff
@@ -260,8 +260,9 @@ class ChatMessage {
...
       return codes[code]
```
Comment (wvffle) [inline lib/chat_message.js:266]: Separate `return` from if statements with `\n`

File: lib/chat_message.js [hunk 4]
```diff
@@ -273,7 +274,9 @@ class ChatMessage {
       else message += vsprintf(format, args)
     }
     if (this.extra) {
-      message += this.extra.map((entry) => entry.toMotd(lang)).join('')
+      message += this.extra.map((entry, index) => {
+        return entry.toMotd(lang, this)
+      }).join('')
```
Comment (wvffle) [inline lib/chat_message.js:279]: `index` is not used

File: lib/chat_message.js [hunk 5]
```diff
@@ -274,9 +274,7 @@ class ChatMessage {
       else message += vsprintf(format, args)
     }
     if (this.extra) {
-      message += this.extra.map((entry, index) => {
-        return entry.toMotd(lang, this)
-      }).join('')
+      message += this.extra.map(entry => entry.toMotd(lang, this)).join('')
```
Comment (Nurutomo) [inline lib/chat_message.js:277]: Removed unused args

File: lib/chat_message.js [hunk 6]
```diff
@@ -260,9 +260,9 @@ class ChatMessage {
     }
 
     let message = Object.keys(codes).map((code) => {
-      self[code] = parent[code] || this[code]
-      if (!self[code] || self[code] === 'false') return null
-      if (code === 'color') return codes.color[self.color]
+      this[code] = parent[code] || this[code]
+      if (!this[code] || this[code] === 'false') return null
+      if (code === 'color') return codes.color[this.color]
```
Comment (Nurutomo) [inline lib/chat_message.js]: Use `this` instead of `self`

**Comment by rom1504** [comment]
My comment about the unit test is still valid :)

File: lib/chat_message.js [hunk 7]
```diff
@@ -271,9 +272,9 @@ class ChatMessage {
       const format = lang[this.translate]
       if (!format) message += args.join('')
       else message += vsprintf(format, args)
-    }
+    } else if (this.translate) message += lang[this.translate]
```
Comment (Nurutomo) [inline lib/chat_message.js:275]: Translate in advancement dont get translated so, i fix it
Result

File: lib/chat_message.js [hunk 8]
```diff
@@ -260,6 +260,7 @@ class ChatMessage {
     }
 
     let message = Object.keys(codes).map((code) => {
+      this[code] = this[code] || parent[code]
```
Comment (Nurutomo) [inline lib/chat_message.js:263]: swapped `this` and `parent`

**Comment by Nurutomo** [comment]
> My comment about the unit test is still valid :)
Is that means no error?

File: lib/chat_message.js (hunk 7 above)
Comment (wvffle) [inline lib/chat_message.js:275]: Add brackets.

File: lib/chat_message.js [hunk 9]
```diff
@@ -271,9 +274,11 @@ class ChatMessage {
       const format = lang[this.translate]
       if (!format) message += args.join('')
       else message += vsprintf(format, args)
+    } else if (this.translate) {
+      message += lang[this.translate]
```
Comment (Nurutomo) [inline lib/chat_message.js:278]: Now in bracket

File: lib/chat_message.js [hunk 10]
```diff
@@ -221,6 +221,8 @@ class ChatMessage {
...
```
Comment (Nurutomo) [inline lib/chat_message.js:225]: also this

File: lib/chat_message.js (hunk 9 above)
Comment (Nurutomo) [inline lib/chat_message.js:278]: Bracket

File: lib/chat_message.js (hunk 10 above)
Comment (Nurutomo) [inline lib/chat_message.js:225]: Add these to `toString()`

**Comment by rom1504** [comment]
There is no test so I still have no idea if it works, changing a few minor implementation details won't change this

**Comment by Nurutomo** [comment]
Sorry for the commits, because i don't know how to create test

**Comment by Nurutomo** [comment]
Finally!

File: test/externalTests/chat.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+module.exports = (supportedVersion) => (bot, done) => {
+  setTimeout(() => {
+    bot.chat('/tellraw @p {"translate":"language.name"}')
+    bot.once('message', (json) => {
+      const str = json.toString()
+      console.log(str)
+      assert.strictEqual(str, 'English')
+      done()
+    })
```
Comment (Nurutomo) [inline test/externalTests/chat.js:11]: I am too lazy at here :P

**Pull request merged**